File: app/src/main/java/net/ducksmanager/persistence/models/composite/InducksPublicationWithPossession.java
package net.ducksmanager.persistence.models.composite;

import net.ducksmanager.persistence.models.coa.InducksPublication;

import androidx.room.ColumnInfo;
import androidx.room.Embedded;

public class InducksPublicationWithPossession {
    @Embedded
    private InducksPublication publication;

    @ColumnInfo
    private Boolean isPossessed;

    public InducksPublication getPublication() {
        return publication;
    }

    public void setPublication(InducksPublication publication) {
        this.publication = publication;
    }

    public Boolean getPossessed() {
        return isPossessed;
    }

    public void setPossessed(Boolean possessed) {
        isPossessed = possessed;
    }
}

File: app/src/main/java/net/ducksmanager/persistence/models/composite/InducksPublicationWithPossession.kt
package net.ducksmanager.persistence.models.composite

import androidx.room.Embedded
import net.ducksmanager.persistence.models.coa.InducksPublication

class InducksPublicationWithPossession : OwnershipAndReferenceCount() {
    @Embedded
    lateinit var publication: InducksPublication
}